hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/bucket/BucketCache.java

```diff
@@ -1056,6 +1069,7 @@ private void retrieveFromFile(int[] bucketSizes) throws IOException, BucketAlloc
             "Attempt to restore non-persistent cache mappings!");
       fis = new FileInputStream(persistencePath);
       ois = new ObjectInputStream(fis);
+      ois.readUTF();
```

Comment:
It breaks backward compatibility. For an upgrading rs, it should read an old persistent file.